site/app/controllers/pdf/PDFController.php

```diff
@@ -30,24 +30,13 @@ public function showStudentPDF($gradeable_id, $filename, $path) {
         $submitter = $this->core->getQueries()->getSubmitterById($id);
         $graded_gradeable = $this->core->getQueries()->getGradedGradeableForSubmitter($gradeable, $submitter);
         $active_version = $graded_gradeable->getAutoGradedGradeable()->getActiveVersion();
-        $annotation_path = FileUtils::joinPaths($this->core->getConfig()->getCoursePath(), 'annotations', $gradeable_id, $id, $active_version);
+        $annotation_dir = preg_replace('/\\.[^.\\s]{3,4}$/', '', FileUtils::joinPaths($this->core->getConfig()->getCoursePath(), 'annotations', $gradeable_id, $id, $active_version));
         $annotation_jsons = [];
-        if (is_dir($annotation_path) && count(scandir($annotation_path)) > 2) {
-            $first_file = scandir($annotation_path)[2];
-            $annotation_path = FileUtils::joinPaths($annotation_path, $first_file);
-            if (is_file($annotation_path)) {
-                $dir_iter = new \DirectoryIterator(dirname($annotation_path . '/'));
-                foreach ($dir_iter as $fileinfo) {
-                    if (!$fileinfo->isDot()) {
-                        $no_extension = preg_replace('/\\.[^.\\s]{3,4}$/', '', $fileinfo->getFilename());
-                        $pdf_info = explode('_', $no_extension);
-                        $pdf_id = implode('_', array_slice($pdf_info, 0, -1));
-                        $grader_id = $pdf_info[count($pdf_info) - 1];
-                        if ($pdf_id . '.pdf' === $filename) {
-                            $annotation_jsons[$grader_id] = file_get_contents($fileinfo->getPathname());
-                        }
-                    }
-                }
+        if (is_dir($annotation_dir)) {
+            foreach (scandir($annotation_dir) as $annotation_file) {
+                $annotation_decoded = json_decode($annotation_file);
+                $grader_id = $annotation_decoded["userId"];
+                $annotation_jsons[$grader_id] = file_get_contents(FileUtils::joinPaths($annotation_dir, $annotation_file));
```

Comment:
For this, and other loops over the annotation directory, we need to limit to grabbing only annotations for the set filename. The reason for doing a md5 hash of just the file path is then within this loop (and others like it), we can do a compare of `explode('_', $annotation_file)[0] === md5($filename)` or something.

Reply:
This has been added in the latest commit